builtin/credential/aws/client.go

```diff
@@ -80,20 +81,39 @@ func (b *backend) getClientConfig(s logical.Storage, region, stsRole, clientType
 		return nil, fmt.Errorf("could not compile valid credentials through the default provider chain")
 	}
 
+	stsConfig, err := b.getRawClientConfig(s, region, "sts")
```

Comment:
Can you explain this change? Will it always be a valid assumption to assume that you're getting STS credentials?